onnx/backend/test/case/node/rotaryembedding.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import numpy as np
+
+import onnx
+from onnx.backend.test.case.base import Base
+from onnx.backend.test.case.node import expect
+
+
+def compute_rotary_embedding(
```

Comment:
Let us share a common function between here and the reference-implementation if possible. Better than duplicating same logic.